src/js/components/Carousel.js

```diff
@@ -133,29 +140,42 @@ export default class Carousel extends Component {
       const { activeIndex } = this.state;
       const { intl } = this.context;
       const numSlides = children.length;
-
-      this.setState({
-        activeIndex: (activeIndex + 1) % numSlides
-      }, () => {
+      const index = (activeIndex + 1) % numSlides;
+      const announceFunc = () => {
         const slideNumber = Intl.getMessage(intl, 'Slide Number', {
           slideNumber: this.state.activeIndex + 1
         });
         const activatedMessage = Intl.getMessage(intl, 'Activated');
         announce(`${slideNumber} ${activatedMessage}`, 'polite');
-      });
+      };
 
+      if (! this.props.hasOwnProperty('activeIndex')) {
+        this.setState({
+          activeIndex: index
+        }, announceFunc );
+      }
       if (!infinite && activeIndex === numSlides - 1) {
         clearInterval(this._slideAnimation);
       }
+
+      if (this.props.onActive) {
+        this.props.onActive(index);
+        announceFunc();
```

Comment:
:o Right , mm and not. The announceFunc is only called if this is true `! this.props.hasOwnProperty('activeIndex')` If that is false the dev will set the state through `componentWillReceiveProps`, but they cant explicitly call the announceFunc that is why I called the `announceFunc` once the dev has updated the props by itself (activeIndex={..}), fired by `        this.props.onActive(index);`
However my code can lead to call twice `announceFunc`, I think the fix will be:

`if (this.props.onActive) {`
`        this.props.onActive(index);`
`        if (this.props.hasOwnProperty('activeIndex'))
          announceFunc();
      }`